ERR: Fix missing space in warning message

MicaelJarniac opened the pull request:
- [x] closes #39158 
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them

MicaelJarniac committed:
DOC: Fix missing space

MicaelJarniac committed:
Merge branch 'master' into fix-warning-typo

simonjayhawkins commented on pandas/_libs/parsers.pyx:
```diff
@@ -1991,7 +1991,7 @@ def _concatenate_chunks(list chunks):
     if warning_columns:
         warning_names = ','.join(warning_columns)
         warning_message = " ".join([
```
does removing the join here work instead and just use fstring literal concatenation

MicaelJarniac replied:
I can't see why that wouldn't work. Technically, the `join` wasn't even being used anyways, because the strings were getting concatenated inside of the list, so yeah, "officially" removing the `join` and manually adding the space between the strings sounds perfectly fine I think.
I'll poke around to see how other similar warnings are created, though, because keeping it consistent between them might be a valid reason as to whether or not to have the `join`.

simonjayhawkins reviewed (commented):
Thanks @MicaelJarniac for the PR. 

We normally check error messages in the tests so we would normally expect that changing a message would also require a change in the tests.

so it maybe that there are no tests that hit this or the message is only partially checked. so further investigation welcome.

MicaelJarniac commented:
> Thanks @MicaelJarniac for the PR.
> 
> We normally check error messages in the tests so we would normally expect that changing a message would also require a change in the tests.
> 
> so it maybe that there are no tests that hit this or the message is only partially checked. so further investigation welcome.

I'll try locating the tests related to this specific message (if any), and fix (or add) them.

And I'll also try to think of a way for catching similar issues in future tests, but I don't think it'd be easy, as simply saying "fail if no space after period" would catch this one, but would probably have about a thousand false-positives in other messages, because in Python syntax it's perfectly valid to have `some.Thing`. A test for catching issues similar to this one might end up being overkill.

simonjayhawkins commented:
> so it maybe that there are no tests that hit this or the message is only partially checked. so further investigation welcome.

it appears that test_warn_if_chunks_have_mismatched_type[c_low] hits this. so can modify that test to also check message.

MicaelJarniac committed:
Merge branch 'master' into fix-warning-typo

MicaelJarniac committed:
TST: Check if warning message matches

pytest pandas/tests/io/parser/common/test_chunksize.py::test_warn_if_chunks_have_mismatched_type

MicaelJarniac commented on pandas/tests/io/parser/common/test_chunksize.py:
```diff
@@ -181,8 +181,14 @@ def test_warn_if_chunks_have_mismatched_type(all_parsers):
+    if record:
+        assert (
+            str(record[0].message)
+            == "Columns (0) have mixed types. Specify dtype option on import or "
```
The string is quite long, so it doesn't look particularly pretty in here, but I think it's okay-ish.

jbrockmendel replied:
usually we split this into two lines

```
expected = "Columns (0) ..."
assert str(record[0].message) == expected
```

jbrockmendel commented:
@MicaelJarniac can you merge master and we'll try to get this in

MicaelJarniac committed:
TST: Split long message

MicaelJarniac committed:
Merge remote-tracking branch 'upstream/master' into fix-warning-typo

jbrockmendel reviewed (approved):
LGTM

simonjayhawkins commented on pandas/tests/io/parser/common/test_chunksize.py:
```diff
@@ -190,8 +190,14 @@ def test_warn_if_chunks_have_mismatched_type(all_parsers, request):
 
     try:
-        with tm.assert_produces_warning(warning_type):
+        with tm.assert_produces_warning(warning_type) as record:
             df = parser.read_csv(buf)
+        if record:
+            expected = (
+                "Columns (0) have mixed types. Specify dtype option on import or "
+                "set low_memory=False."
+            )
+            assert str(record[0].message) == expected
```
`tm.assert_produces_warning` now accepts a `match` argument

```suggestion
        msg = (
            "Columns (0) have mixed types. Specify dtype option on import or "
            "set low_memory=False."
        )
        with tm.assert_produces_warning(warning_type, match=re.escape(msg)):
            df = parser.read_csv(buf)
```

MicaelJarniac committed:
TST: Use `match` argument

Co-authored-by: Simon Hawkins <simonjayhawkins@gmail.com>

MicaelJarniac committed:
TST: Fix missing regex import

jreback commented:
closing as stale. @MicaelJarniac if you'd like to rebase and fix this up pls ping.
